### ISSUE TITLE
Code blocks inside JSX tags disrupt highlighting

### ISSUE THREAD

**Issue by Josh-Cena** [silvenon/vscode-mdx#22]
Two repros:

````jsx
<Code>

```java
class Hi {
  public static void main(String args[]) {
    console.log("Hi");
  }
}
```

</Code>
````

![image](https://user-images.githubusercontent.com/55398995/135869706-f3cf6025-47f2-4f38-85ac-3e5e69125912.png)

````jsx
<details>

```bash
website/<locale>/docs
```

</details>

# Title
````

![image](https://user-images.githubusercontent.com/55398995/135869762-2480c53b-ec05-4c4a-8c1a-aed4d960d782.png)

**Comment by favna** [comment]
Commenting to follow this issue because I'm facing the problem as well:

![image](https://user-images.githubusercontent.com/4019718/138560350-76f2d408-adfc-4b08-8a56-40d6a2ecfe71.png)

**Comment by Josh-Cena** [comment]
@Favna If you think this has affected you significantly, you may want to wrap the entire Tabs in `mdx-code-block`: https://docusaurus.io/docs/i18n/crowdin#mdx-solutions This is a really useful escape hatch for multiple issues, including this one. Remember to use four backticks though

**Comment by favna** [comment]
I just tried it in 1 document, but that removes all syntax highlighting except that of headers. It's not really much better IMHO.

**Comment by Josh-Cena** [comment]
> I just tried it in 1 document, but that removes all syntax highlighting except that of headers. It's not really much better IMHO.

Don't wrap the entire document, just the broken `<Tabs>` component. You do lose highlighting for that comp, but you still have highlighting for the rest.

**Comment by pomber** [comment]
Isn't this the same problem as #14? 

I'm very interested in a solution. I know nothing about textmate grammars but let me know if I can help.

**Comment by Josh-Cena** [comment]
Yes, they are caused by the same reason, but are manifested differently and have different impacts IMO. In their case, the inline code is not getting any highlight whatsoever, while in this case, it's getting highlighted, but in a very incorrect way (the content is seen as JSX rather than MD) which prevents any succeeding text from being parsed correctly. If possible, I'd rather have the text between JSX tags treated as plain text with no highlighting at all.

**Comment by Josh-Cena** [comment]
I have some experience with textmate but considering the current status of MDX (as you mentioned in #14) and how occupied I've been recently, I would wait for a few months before forking this and releasing my own extension if I got no response XD
